arangod/Cluster/ServerState.cpp

```diff
@@ -385,7 +411,15 @@ bool ServerState::integrateIntoCluster(ServerState::RoleEnum role,
                                     << roleToString(role) << " and our id is " << id;
 
   // now overwrite the entry in /Current/ServersRegistered/<myId>
-  return registerAtAgencyPhase2(comm);
+  if (!registerAtAgencyPhase2(comm)) {
+    return false;
+  }
+
+  // check that our rebootId was written
+  setRebootId(readRebootId(comm));
+  TRI_ASSERT(_rebootId > 0);
```

Comment:
I think this is not save for production (see comment above)
In case readRebootId returns `0` => error case
We need to `return false` here  and let the server crash with invalid state.
We can implement retry around the reads in addition.